src/ctapipe/irf/benchmarks.py

```diff
@@ -0,0 +1,306 @@
+    ) -> BinTableHDU:
+        if self.use_true_energy:
+            e_bins = self.true_energy_bins
+            energy_type = "true"
+        else:
+            e_bins = self.reco_energy_bins
+            energy_type = "reco"
+
+        result, fov_bin_idx, mat_shape = _get_2d_result_table(
+            events=events,
+            e_bins=e_bins,
+            fov_bins=self.fov_offset_bins,
+        )
+        result["N_EVENTS"] = np.zeros(mat_shape)[np.newaxis, ...]
+        for q in self.quantiles:
+            result[f"ANGULAR_RESOLUTION_{q * 100:.0f}"] = u.Quantity(
+                np.full(mat_shape, np.nan)[np.newaxis, ...], events["theta"].unit
+            )
+
+        for i in range(len(self.fov_offset_bins) - 1):
+            ang_res = angular_resolution(
+                events=events[fov_bin_idx == i],
+                energy_bins=e_bins,
+                energy_type=energy_type,
+                quantile=self.quantiles,
+            )
+            result["N_EVENTS"][:, i, :] = ang_res["n_events"]
+            for q in self.quantiles:
+                result[f"ANGULAR_RESOLUTION_{q * 100:.0f}"][:, i, :] = ang_res[
+                    f"angular_resolution_{q * 100:.0f}"
+                ]
+
+        header = Header()
+        header["E_TYPE"] = energy_type.upper()
+        return BinTableHDU(result, header=header, name=extname)
+
+
+class SensitivityMakerBase(DefaultRecoEnergyBins):
+    """Base class for calculating the point source sensitivity."""
+
+    alpha = Float(
+        default_value=0.2,
+        help="Size ratio of on region / off region.",
+    ).tag(config=True)
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+
+    @abstractmethod
+    def __call__(
+        self,
+        signal_events: QTable,
+        background_events: QTable,
+        spatial_selection_table: QTable,
+        gamma_spectrum: Spectra,
+        extname: str = "SENSITIVITY",
+    ) -> BinTableHDU:
+        """
+        Calculate the point source sensitivity
+        based on ``pyirf.sensitivity.calculate_sensitivity``.
+
+        Parameters
+        ----------
+        signal_events: astropy.table.QTable
+            Reconstructed signal events to be used.
+        background_events: astropy.table.QTable
+            Reconstructed background events to be used.
+        spatial_selection_table: QTable
+            Direction cut that was applied on ``signal_events``.
+        gamma_spectrum: ctapipe.irf.Spectra
+            Spectra by which to scale the relative sensitivity to get the flux sensitivity.
+        extname: str
+            Name of the BinTableHDU.
+
+        Returns
+        -------
+        BinTableHDU
+        """
+
+
+class Sensitivity2dMaker(SensitivityMakerBase, DefaultFoVOffsetBins):
+    """
+    Calculates the point source sensitivity in bins of reconstructed energy
+    and fov offset.
+    """
```

Comment:
It's not obvious from the code, but does this class use the create_bins_per_decade function create here or the one from pyirf?

To comply with the cta requirements, we should use the one that produces _exactly_ five bins per decade.